webapp/src/index.tsx

```diff
@@ -22,8 +22,14 @@ export default class Plugin {
         const hooks = new Hooks(store);
         registry.registerSlashCommandWillBePostedHook(hooks.slashCommandWillBePostedHook);
 
-        const setup = () => {
+        const setup = async () => {
+            // Retrieve provider configuration so we can access names and other options in messages to use in the frontend.
+            await store.dispatch(getProviderConfiguration());
```

Comment:
We also want to move the `registerChannelHeaderMenuAction` into this `setup` function, unless we're returning null